### DISCUSSION TITLE
Portrait format.

### DISCUSSION THREAD

**Discussion by Kot-Nikodim** [richardwilkes/gcs#654] [Q&A] [answered]
Hi. I try to make simple reader for *.gcs files on java. How do you save images? Old version used base64. But I can't decode images from new versions.

**Comment by richardwilkes** [comment, marked as answer]
The data inside the base64 encoding is the image format, which depends on what the user used when they added the image. GCS supports WEBP, PNG, JPEG, GIF, BMP, WBMP, and ICO as inputs. If the image isn't the correct size, it will be rescaled and turned into a WEBP, otherwise it will be left as-is.

**Comment by Kot-Nikodim** [comment]
So. Do all versions use base64?

**Comment by richardwilkes** [comment]
Yes, I believe so. There may have been some special case for the default portrait, but I don't currently have the spare time to check. The source is available to look at if you're interested, though.
